swatch-core/src/main/java/org/candlepin/subscriptions/db/model/EventRecord.java

```diff
@@ -91,8 +92,23 @@ public EventRecord(Event event) {
   @Column(name = "instance_id")
   private String instanceId;
 
+  /*
+  Since we have a bitemporal pattern, the "timestamp" and "actual_date" means the same.
+  Accordingly, "record_date" refers to the date when we entered the activity into our system,
+  "update_date" refers to the date when we made any changes to the record.
+
+  For reference: https://martinfowler.com/articles/bitemporal-history.html#TheTwoDimensions
+  */
+
   private OffsetDateTime timestamp;
 
+  @CreationTimestamp
+  @Column(name = "record_date")
+  private OffsetDateTime recordDate;
+
+  @Column(name = "update_date")
```

Comment:
If we use the `@UpdateTimestamp` annotation, we would not need to add methods for doing so (speaking about `updateLastSeenTallyEvents`).

Reply:
As we spoke.
Regarding the update timestamp, I have purposely not used that annotation. 
Let me clarify since we don't currently interact with event table for saving during hourly tally, hence the `save all` is not included in the hourly code flow. Previously, you saw the "save all" function included in my PR to handle the flow for saving events by Prometheus or cost management, making it beneficial the next time an hourly tally is done.
Secondly, we are not updating the `update_date` when we save events. That timestamp is like a flag that is used by hourly tally.

This is a marker solely used for hourly tally. I named the column since the Jira card had it, but we can change the name to something other like `last_seen_tally` or anything similar. The issue with the name last_seen_tally is that it will once more be mistaken with the host `last_seen` field. I don't mind changing it to `last_seen_tally` if that solves your confusion.

Reply:
@kahowell thoughts on keeping the same name `update_date` or changing it to some other name?

Reply:
We removed the use of update_date.